scipy/optimize/_optimize.py

```diff
@@ -2179,11 +2178,9 @@ def terminate(warnflag, msg):
                     Ap = fhess_p(xk, psupi, *args)
                     hcalls += 1
             else:
-                if isinstance(A, HessianUpdateStrategy):
-                    # if hess was supplied as a HessianUpdateStrategy
-                    Ap = A.dot(psupi)
-                else:
-                    Ap = np.dot(A, psupi)
```

Comment:
I could not notice a performance difference between `np.dot(a, b` and `a.dot(b)` locally. `a.dot(b)` works for sparse and dense arrays so I thought this would simplify things.